hedera-node/docs/proposals/block-stream-bucket-uploader.md

```diff
@@ -0,0 +1,398 @@
+      "secretKey": "YOUR_GCP_SECRET_KEY"
+    }
+  }
+}
+```
+
+#### Credentials Records
+```java
+public record BucketCredentials(
+    String accessKey,
+    String secretKey
+) {
+    public BucketCredentials {
+        Objects.requireNonNull(accessKey, "Access key cannot be null");
+        Objects.requireNonNull(secretKey, "Secret key cannot be null");
+    }
+}
+
+public record BucketCredentialsConfig(
+    Map<String, BucketCredentials> credentials
+) {
+    public BucketCredentialsConfig {
+        Objects.requireNonNull(credentials, "Credentials map cannot be null");
+    }
+}
+```
+
+#### Configuration Loading and Updates
+The configuration system needs to handle both network properties and local credentials in a secure and maintainable way.
+
+**BucketConfigurationManager Requirements**
+- Singleton class managing both network properties and local credentials
+- Load credentials from the JSON file specified in the configuration
+- Provide methods to get complete bucket configurations by combining network properties with credentials
+- Handle credential file reload if needed
+- Log appropriate warnings/errors when credentials are missing or invalid
+- Validate that all required bucket configurations have matching credentials
+
+**CompleteBucketConfig Requirements**
+- Immutable record combining network configuration with credentials
+- Include all necessary fields for bucket connection:
+  - name
+  - provider type (AWS/GCP)
+  - endpoint
+  - region (if applicable)
+  - bucket name
+  - credentials (access key and secret key)
+- Include validation to ensure all required fields are present
+- Provide a clean interface for the MinioClientFactory to create new clients
+
+### Cloud Storage Interface and Implementations
+
+#### CloudBucketUploader Interface
+The base interface defining operations for uploading blocks to cloud storage providers.
+
+Key Operations:
+- Upload a block file with associated metadata asynchronously
+- Check if a block exists in cloud storage
+- Retrieve MD5 hash of a stored block
+- Get the provider identifier
+
+#### MinioBucketUploader Implementation
+Implementation of CloudBucketUploader for MinIO/S3-compatible storage.
+
+Key Responsibilities:
+- Initialize MinIO client with provider configuration
+- Handle block uploads with retry logic and MD5 validation
+- Check for existing blocks and validate hashes
+- Manage concurrent uploads using executor service
+- Format object keys using standardized naming
+- Handle provider-specific errors and exceptions
+
+Required Features:
+- Configurable retry attempts for failed uploads
+- Exponential backoff with jitter between retries
+- Hash validation before and after uploads
+- Asynchronous operations using CompletableFuture
+- Structured logging of operations and errors
+
+#### RetryUtils Helper
+Utility class providing standardized retry logic for operations.
+
+Key Features:
+- Configurable maximum retry attempts
+- Exponential backoff with random jitter
```

Comment:
It feels like we don't need this sentence repeated again here, could maybe remove from `Required Features`?